### PR TITLE
Onboarding: Get rid of the headstart

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/wp-calypso/issues/80732, D119166-code

## Proposed Changes

* As the title, we want to get rid of the headstart during the onboarding. Hence, this PR is focusing on avoid calling the theme-setup endpoint during the onboarding

## Testing Instructions

* Go to /setup?siteSlug=<your_site>
* Continue until you land on the Design Picker
* Pick the following themes that have a static homepage
  * byrne
  * arbutus
  * attar
  * geologist
  * marl
* Ensure the homepage after the onboarding is the same as the theme
* Repeat the above steps, and now pick the themes that are using the template as their homepage
* Ensure the homepage is still correct after the onboarding

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by taipeicoder — APPROVED** [review]
LGTM 🎉 It'd be great if you could write a post about these changed after they've been merged 🙂

File: packages/data-stores/src/site/actions.ts
Comment (okmttdhr) [inline packages/data-stores/src/site/actions.ts]: badge `/non-blocker`
Since it's a special case, we might feel more comfortable putting it to the test. ✅

**Review by okmttdhr — APPROVED** [review]
Awesome 🙌 Tested and LGTM!

File: packages/data-stores/src/site/actions.ts
Comment (arthur791004) [reply packages/data-stores/src/site/actions.ts]: Updated!

**Comment by arthur791004** [comment]
Reverted via https://github.com/Automattic/wp-calypso/pull/80850

**Comment by arthur791004** [comment]
Proposing the [second try](https://github.com/Automattic/wp-calypso/pull/80851)

**Pull request merged**
